diff --git a/mt7925/mac.c b/mt7925/mac.c
index 901ad5d..a0b71f8 100644
--- a/mt7925/mac.c
+++ b/mt7925/mac.c
@@ -752,6 +752,10 @@ mt7925_mac_write_txwi(struct mt76_dev *dev, __le32 *txwi,
 		omac_idx = mvif->omac_idx;
 		wmm_idx = mvif->wmm_idx;
 		band_idx = mvif->band_idx;
+	} else if (vif && vif->type == NL80211_IFTYPE_MONITOR) {
+		if (dev->mt76.chandef.chan &&
+		    dev->mt76.chandef.chan->band != NL80211_BAND_2GHZ)
+			band_idx = 1;
 	}
 
 	if (inband_disc) {
